### PR TITLE
Added Matrix4 pybind11 interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

This PR creates a Matrix4 Pybind11 interface

Require
 - https://github.com/ignitionrobotics/ign-math/pull/334

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by chapulina — CHANGES_REQUESTED** [review]
I think we should address this TODO here:

https://github.com/ignitionrobotics/ign-math/blob/f6c26bfc32b730a4a18b735ba492eca53188b186/src/python_pybind11/test/Quaternion_TEST.py#L358-L364

File: src/python_pybind11/src/_ignition_math_pybind11.cc [hunk 1]
```diff
@@ -95,6 +96,10 @@ PYBIND11_MODULE(math, m)
   ignition::math::python::defineMathQuaternion<double>(m, "Quaterniond");
   ignition::math::python::defineMathQuaternion<float>(m, "Quaternionf");
 
+  ignition::math::python::defineMathMatrix4<int>(m, "Matrix4i");
+  ignition::math::python::defineMathMatrix4<double>(m, "Matrix4d");
+  ignition::math::python::defineMathMatrix4<float>(m, "Matrix4f");
```
Comment (chapulina) [inline src/python_pybind11/src/_ignition_math_pybind11.cc:101]: Ah and this could go up close to `Matrix3`

**Review by chapulina — APPROVED** [review]
LGTM with happy CI

File: src/python_pybind11/src/Matrix4.hh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+         "Get the scale values as a Vector3<T>")
+    .def("transform_affine",
+         py::overload_cast<const ignition::math::Vector3<T>&,
+                           ignition::math::Vector3<T>&>(
+                             &Class::TransformAffine, py::const_),
+         "Perform an affine transformation")
+    .def("equal",
+         &Class::Equal,
+         "Equality test operator")
+    .def("look_at",
```
Comment (chapulina) [inline src/python_pybind11/src/Matrix4.hh:123]: The test failed:

```
   ERROR: test_look_at (__main__.TestMatrix4)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/github/workspace/src/python_pybind11/test/Matrix4_TEST.py", line 450, in test_look_at
      self.assertAlmostEqual(Matrix4d.look_at(-Vector3d.UNIT_X,
  TypeError: look_at(): incompatible function arguments. The following argument types are supported:
      1. (self: ignition.math.Vector3d, arg0: ignition.math.Vector3d, arg1: ignition.math.Vector3d) -> ignition.math.Matrix4d
  
  Invoked with: -1 -0 -0, 0 0 0
```

**Review by chapulina — CHANGES_REQUESTED** [review]

**Review by chapulina — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**